polar-core/src/vm.rs

```diff
@@ -1418,6 +1425,27 @@ impl PolarVirtualMachine {
                                 .collect::<Vec<Goals>>(),
                         )?;
                     }
+                    // Push an `ExternalLookup` goal for external instances and built-ins.
+                    Value::Dictionary(_) | Value::ExternalInstance(_) | Value::String(_) => {
+                        // Generate symbol for next result and bind to `false` (default)
+                        let next = self.kb.read().unwrap().gensym("next_value");
+                        self.bind(&next, Term::new_temporary(Value::Boolean(false)));
+                        let next_term = Term::new_temporary(Value::Variable(next.clone()));
+                        let call_id = self.new_call_id(&next);
+
+                        // append unify goal to be evaluated after
+                        // next result is fetched
+                        self.append_goals(vec![
+                            Goal::NextExternal {
+                                call_id,
+                                term: self.deep_deref(&list),
```

Comment:
I think we should rename `list` to reflect that it could be a non-list. Maybe `iterator` would be better.

Reply:
Done